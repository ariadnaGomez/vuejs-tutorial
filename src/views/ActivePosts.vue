<template>
  <div class="posts-list">
    <h1>Posts activos</h1>
    <router-link to="/posts">Go back</router-link>
    <div
      v-for="post in posts"
      :key="post.id"
      :id="post.id"
      class="post"
    >
      <p>{{ post.title }}</p>
      <p>{{ post.body }}</p>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  name: 'Posts',
  computed: mapGetters({
    posts: 'activePosts'
  })
}
</script>

<style lang="scss">
  .posts-list {
    margin: 0 auto;
    max-width: 700px;
    .post {
      border: 1px solid blue;
      margin: 20px;
      padding: 20px;
    }
    p {
      text-transform: capitalize;
    }
  }
</style>
