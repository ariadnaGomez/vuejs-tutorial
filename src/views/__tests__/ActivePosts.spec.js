import { mount, createLocalVue } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Vuex from 'vuex'
import ActivePosts from '@/views/ActivePosts.vue'

describe('Posts', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  let store
  let getters
  beforeEach(() => {
    getters = {activePosts: jest.fn(() => [
      {id: 3, isActive: true, title: 'title2'}])}
    store = new Vuex.Store({
      getters
    })
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = mount(ActivePosts, {localVue, store})
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    getters.activePosts.mockImplementation(() => [
      {id: 3, isActive: true, title: 'title2'}, {id: 4, title: 'title3'}])
    const wrapper = mount(ActivePosts, {localVue, store})
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
