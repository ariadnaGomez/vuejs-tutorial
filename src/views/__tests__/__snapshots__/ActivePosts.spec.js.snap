// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts has same HTML structure 1`] = `"<div data-server-rendered=\\"true\\" class=\\"posts-list\\"><h1>Posts activos</h1> <router-link to=\\"/\\">Go back</router-link> <div id=\\"3\\" class=\\"post\\"><p>title2</p> <p></p></div></div>"`;

exports[`Posts has same HTML structure 2`] = `"<div data-server-rendered=\\"true\\" class=\\"posts-list\\"><h1>Posts activos</h1> <router-link to=\\"/\\">Go back</router-link> <div id=\\"3\\" class=\\"post\\"><p>title2</p> <p></p></div><div id=\\"4\\" class=\\"post\\"><p>title3</p> <p></p></div></div>"`;
